src/BinaryFormatter/Utils/TypeHelper.cs

```diff
@@ -24,5 +26,45 @@ public static class TypeHelper
             }
             throw new ArgumentException(" ### -> public static KeyValuePair<object , object > CastFrom(Object obj): Error: obj argument must be KeyValuePair <,> ");
         }
+
+        public static bool IsDictionary(object obj)
+        {
+            return obj is IDictionary;
+        }
+
+        public static bool IsList(object obj)
+        {
+            return obj is IList;
+        }
+
+        public static bool IsLinkedList(object obj)
+        {
+            Type[] genericTypes = obj.GetType().GenericTypeArguments;
+
+            if (genericTypes.Length == 1)
+            {
+                var listType = typeof(LinkedList<>);
+                var constructedListType = listType.MakeGenericType(genericTypes);
+                return HasConversionOperator(obj.GetType(), constructedListType);
+            } else
```

Comment:
i guess else statement is redundant

Reply:
@lukasz-pyrzyk Agree. FIx in the commit 11163ed